### PR TITLE
HDM defaults spec

### PR DESCRIPTION
**Pull request by jeromekelleher**
Some basic text for the HDM defaults section. I think this is probably enough: this is to teach people how to use it, just to explain to parser developers how it should be implemented. The combination of the schema and reference implementation is pretty thorough, so I think we don't have to do so much here.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
(The changes in the reference impl is to just move the entry point function to the top)

File: docs/specification.md [hunk 1]
```diff
@@ -410,17 +417,93 @@ This section defines the structure of HDM documents, the rules by
...
+(sec_spec_hdm_defaults_top_level)=
+#### Top-level defaults
+
+The top-level HDM document may contain a propery ``defaults``,
+which defines values to use for entities within rest of the document
+*unless otherwise specified*. The ``defaults`` object can have
+the following properties:
+
+- epoch: this can specify any property valid for an
+  {ref}`MDM epoch<sec_spec_mdm_epoch>`. Any epochs in the
```
Comment (grahamgower) [inline docs/specification.md:462]: s/Any/All/ ? Likewise for the migration and pulse paragraphs.

**Review by grahamgower — APPROVED** [review]
Nice! I just had a small language nitpick, which you're free to ignore.

File: docs/specification.md (hunk 1 above)
Comment (jeromekelleher) [reply docs/specification.md]: Done

**Pull request merged**